radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBookEntry.java

```diff
@@ -121,8 +122,8 @@ public static AddressBookEntry createBanned(NodeId nodeId, Instant bannedUntil)
 		return new AddressBookEntry(nodeId, Optional.of(bannedUntil), ImmutableSet.of());
 	}
 
-	public static AddressBookEntry create(RadixNodeUri uri, Instant lastSuccessfulConnection) {
-		return create(uri, Optional.of(lastSuccessfulConnection));
+	public static AddressBookEntry createWithLatestConnectionStatus(RadixNodeUri uri, LatestConnectionStatus latestConnectionStatus) {
+		return create(uri, Optional.of(latestConnectionStatus));
```

Comment:
Maybe Optional.ofNullable?

Reply:
I think we should only use `ofNullable` when the null value is an expected valid case (like, for example, the `deserialize` method that has to deal with previously persisted entries). In general we shouldn't expose any nulls (or methods that accept nulls) outside the local scope.

Reply:
We can't prevent user from passing null. The way it is now a null pointer exception is gonna be thrown. Do we want to do that or fallback to Optional.empty()?

Reply:
I'd rather have an NPE in this case, we just shouldn't be using nulls

Reply:
Maybe it is a good idea. Should we adopt this approach for the whole codebase?

Reply:
we already do that, other than a few local uses and some interfaces that are explicitly marked with `@Nullable`

Reply:
This class uses `Optional.ofNullable` at line 307.

Reply:
ah probably it is cause Optional is serialised as null.

Reply:
yup, in this case receiving null is expected, hence `ofNullable`